### DISCUSSION TITLE
LotW Grid squares are not pulling in during LotW Sync - Intended?

### DISCUSSION THREAD

**Discussion by T012m3n7oR** [magicbug/Cloudlog#1059] [Q&A] [answered]
This is in reference to a previous question #1004 

Is this intended to be added?

In the [/application/controllers/Lotw.php:674](https://github.com/magicbug/Cloudlog/blob/656ff24744e095a6bfc2b3fc8d9bdff5fd4e26ed/application/controllers/Lotw.php#L674) I can see there is a TODO to add an option for this, but I don't know if this is still planned as the TODO was added to the software it looks like back in 2013.

I can see from the [/application/controllers/Lotw.php:443](https://github.com/magicbug/Cloudlog/blob/656ff24744e095a6bfc2b3fc8d9bdff5fd4e26ed/application/controllers/Lotw.php#L443) (loadFromFile) that this calls on line 516 the [lotw_update](https://github.com/magicbug/Cloudlog/blob/656ff24744e095a6bfc2b3fc8d9bdff5fd4e26ed/application/controllers/Lotw.php#L516) function on the logbook_model.  This function does not import the Grid Square that exists on the LotW record.

The lotw_update function exists in [application/models/Logbook_model.php:1374](https://github.com/magicbug/Cloudlog/blob/656ff24744e095a6bfc2b3fc8d9bdff5fd4e26ed/application/models/Logbook_model.php#L1374).  This function does update the database table TABLE_HRD_CONTACTS_V01 with the related QSL data, beside the Grid Squares.

If you want an additional contributor I'm willing to give it a shot - I haven't used PHP before but tracking this down wasn't that hard.

**Comment by magicbug** [comment, marked as answer]
Yeah it's currently intentional, probably not a huge issue on HF but on VHF+ it's seen quite often the operator uploads qsos against the wrong square and I hadn't decided how best to handle it beyond not updating the fields if data is already present

**Reply by T012m3n7oR** [reply]
I can take a look at this later and see if I can come up with an elegant solution.  Will probably be based off options though, because personally the map above the log is one of the primary reasons I chose cloudlog, because that was cool, and it did everything else I needed, and wasn't tied to my personal machine.  But with the map only loading off gridsquare the lack of pulling the gridsquare back hurts as an HF operator primarily for POTA when everything displays in Wisconsin.

**Reply by magicbug** [reply]
Best solution is probably grab the square and add to qso when its not supplied, worth noting LoTW probably return both gridsquare and vucc gridsquare (When you work gridlines) there's two fields in Cloudlog for this as well.

Adding it shouldn't be too bad, will be similar to the US county and I think we pull in state if not probably need todo that too.

**Reply by ofadam** [reply]
U.S. states are pulled only when confirmed by LoTW... not sure if there's a way to pull it before it's confirmed? If so, that would be fantastic so I know which states I've contacted but haven't been able to make official.

**Comment by T012m3n7oR** [comment]
Grid square is written and tested.  Working great, will upload for PR hopefully by next weekend.  Currently written to only import gridsquare if not populated, or if gridsquare matches (LotW has 6/8 characters, and Cloudlog has 4 or 6.)  Vucc Grid Squares I don't have any samples to review to see how this should populate in the database - so I will not be attempting that at this time.

**Reply by T012m3n7oR** [reply]
application\controllers\Lotw.php and application\models\Logbook_model.php updated.

Code slightly refactored in the lotw_update function to remove duplicate code.

**Reply by T012m3n7oR** [reply]
Although 8 digit gridsquares don't seem to be giving expected output.  I have a Canadian gridsquare showing up in the middle of the Golf of Guinea; that is a different issue than what this will be adding though.
